src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle',
+  color: 'red' | 'green' | 'blue',
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: 'triangle';
 
-}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (!this.isValidTriangle()) {
+      throw new Error('Invalid triangle sides');
+    }
+  }
+
+  private isValidTriangle(): boolean {
+    const maxSideA = this.b + this.c > this.a;
+    const maxSideB = this.a + this.c > this.b;
+    const maxSideC = this.a + this.b > this.c;
 
-export class Circle {
+    return maxSideA && maxSideB && maxSideC;
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    const areaTriangle = Math
+      .sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return +areaTriangle.toFixed(2);
```

Comment:
Here you can reuse function that you will create(check my comment on `Circle` class)